### DISCUSSION TITLE
Unable to use URL or GraphQL Endpoint as schema

### DISCUSSION THREAD

**Discussion by edramos-97** [dotansimha/graphql-code-generator#8009] [Q&A] [answered]
Hi all!
So I want to generate types for my frontend queries using the schema from my backend server.
I followed this official [guide](https://www.graphql-code-generator.com/docs/guides/react) since I'm also using react.
I was unable to get the schema URL working. Tried many different real and dummy URLs, none worked and the output message was always the same.

```
Failed to load schema from http://localhost:3002/graphql:

        Request with GET/HEAD method cannot have body.
        TypeError: Request with GET/HEAD method cannot have body.
    at new Request (node_modules\undici\lib\fetch\request.js:437:13)
    at Agent.fetch (node_modules\undici\lib\fetch\index.js:114:21)
    at fetch (node_modules\undici\index.js:90:22)
    at fetch (node_modules\cross-undici-fetch\dist\create-node-ponyfill.js:130:16)
    at fetch (node_modules\cross-undici-fetch\dist\create-node-ponyfill.js:125:18)
    at defaultAsyncFetch (node_modules\@graphql-tools\url-loader\cjs\defaultAsyncFetch.js:6:43)
    at node_modules\@graphql-tools\url-loader\cjs\index.js:212:36
    at new ValueOrPromise (node_modules\value-or-promise\build\main\ValueOrPromise.js:14:21)  
    at executor (node_modules\@graphql-tools\url-loader\cjs\index.js:186:20)
    at node_modules\@graphql-tools\url-loader\cjs\index.js:504:35
```

Tried using different versions of "@graphql-codegen/cli" to no avail, the issue is present from version 2.0.0

The error makes sense and is understandable but I don't think there's any configuration I can setup to prevent it

I'll gladly open a bug report if other people confirm this is an actual problem and not just some misconfiguration on my machine.
I find it hard to believe no one else is using this feature or have previously reported it (couldn't find anything on issues or discussions)

Thanks in advance!

My Environment:
Windows 10
NodeJS: 16.13.2
npm: 8.5.5

**Comment by lewebsimple** [comment]
Having the same problem !

**Reply by edramos-97** [reply]
Hi @lewebsimple thanks for your feedback!
You can also upvote this discussion on the original post so it gets more visibility! 😃

**Reply by lewebsimple** [reply]
Should we create an issue ?

**Reply by saihaj** [reply]
Can you please create an issue with a reproduction for this? I have tried with `@graphql-codegen/cli@2.6.2` in a project and didn't get this error.

**Reply by lewebsimple** [reply]
See here: https://stackblitz.com/edit/node-uxtpcy
`npm run codegen` produces the error.

**Reply by edramos-97** [reply]
Thanks for your input @saihaj 
Created issue #8012

**Comment by edramos-97** [comment, marked as answer]
Elevated to Issue #8012
